src/DatePicker/index.tsx

```diff
@@ -1,33 +1,106 @@
 import * as React from 'react'
-import generatePicker from 'antd/es/date-picker/generatePicker'
+import { default as generatePicker, PickerTimeProps,PickerDateProps , RangePickerTimeProps,RangePickerDateProps } from 'antd/es/date-picker/generatePicker'
 import en from 'antd/es/date-picker/locale/en_US'
-import { CaretDown } from 'phosphor-react'
+import classNames from 'classnames'
+import { ArrowRight, CaretDown } from 'phosphor-react'
 import dateFnsGenerateConfig from 'rc-picker/lib/generate/dateFns'
 
 import './style.css'
 en.lang.quarterFormat = "'Q'Q"
 
 const CaretDownIcon = <CaretDown />
-const DatePicker = generatePicker<Date>(dateFnsGenerateConfig)
+const BaseDatePicker = generatePicker<Date>(dateFnsGenerateConfig)
 
-export const QuarterPicker = DatePicker.QuarterPicker
-export const WeekPicker = DatePicker.WeekPicker
-export const MonthPicker = DatePicker.MonthPicker
-export const YearPicker = DatePicker.YearPicker
-export const RangePicker = DatePicker.RangePicker
-export const TimePicker = DatePicker.TimePicker
+export type TimePickerProps = PickerTimeProps<Date>  & {
+  RangePicker: typeof RangePicker;
+};
+export type TimeRangePickerProps = RangePickerTimeProps<Date> & {
+  popupClassName?: string; 
+}
+export type DateRangePickerProps = RangePickerDateProps<Date>;
+export type DatePickerProps = PickerDateProps<Date>;
 
-QuarterPicker.defaultProps = {
-  format: "YYYY-'Q'Q",
-  suffixIcon: CaretDownIcon
+export const WeekPicker = BaseDatePicker.WeekPicker
+export const MonthPicker = BaseDatePicker.MonthPicker
+export const YearPicker = BaseDatePicker.YearPicker
+export const RangePicker = BaseDatePicker.RangePicker
+
+
+const createHandleOpenChange = (onOpenChange:undefined | ((open:boolean)=>void),setOpen:(open:boolean)=>void) => (isOpen:boolean) =>{
+  setOpen(isOpen);
+  if(onOpenChange){
+    onOpenChange(isOpen);
+  }
+}
+
+const getStyle = (open:boolean,className?:string,) => {return classNames(className,open ? "ui-open" : "")};
+
+export const DatePicker:React.FC<DatePickerProps> = ({ onOpenChange ,className, ...restProps }) => {
+  const [open, setOpen] = React.useState(false);
+  const handleOpenChange = createHandleOpenChange(onOpenChange,setOpen);
+  return(
+    <BaseDatePicker className={getStyle(open,className)} onOpenChange={handleOpenChange} {...restProps}/>
+  );
+}
+
+export const QuarterPicker : React.FC<DatePickerProps> = ({ onOpenChange ,className, ...restProps }) =>{
+  const [open, setOpen] = React.useState(false);
+  const handleOpenChange = createHandleOpenChange(onOpenChange,setOpen);
+  return(
+    <BaseDatePicker className={getStyle(open,className)} onOpenChange={handleOpenChange} {...restProps} picker='quarter' format={"YYYY-'Q'Q"}/>
```

Comment:
To improve performance, you can wrap the style computation with `useMemo` and the open handler with `useCallback`

```suggestion
  const styles = React.useMemo(() => getStyle(open, className), [open, className])
  const handleOpenChange = React.useCallback(createHandleOpenChange(onOpenChange, setOpen), [onOpenChange, setOpen])

  return <BaseDatePicker className={styles} onOpenChange={handleOpenChange} {...restProps} />
```